core/src/main/java/io/parsingdata/metal/data/ParseState.java

```diff
@@ -30,49 +31,59 @@
 import io.parsingdata.metal.encoding.Encoding;
 import io.parsingdata.metal.expression.value.ValueExpression;
 import io.parsingdata.metal.token.Token;
+import javafx.util.Pair;
```

Comment:
JavaFX dependency? I'd rather opt to implement a pair in Metal (I though I read somewhere it already existed? Or non generic, could generify?). This FX implementation is also not a true immutable implementation.

Reply:
Agreed, let's not depend on a random `javafx` class.